### PR TITLE
ISSUE-209: Encode keys before storing on mongo.

### PR DESCRIPTION
**Pull request by samt2497**
# Issue: https://github.com/perftools/xhgui/issues/209

# Notes:
- Added encoding on the profile ( replace . with it Unicode equivalent )

### CODE REVIEW

**Comment by markstory** [comment]
Looks good. I will get the build fixed for php7.

**Comment by lauripiisang** [comment]
It worries me that for me, everything works with mongodb 3.0 (see issue #209 ).

I would like to set up a docker env to test this and find out what the real cause of this issue is before we hurry to a fix that might not work. If you need to use your fix, you can do so via https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository

File: src/Xhgui/Profiles.php [hunk 1]
```diff
@@ -253,6 +253,32 @@ public function getAll($options = array())
...
+    {
+        if (!is_array($profile)) {
+            return $profile;
+        }
+        $target = array(
+          '__encoded' => true,
+        );
+        foreach ($profile as $k => $v) {
+            if (is_array($v)) {
+                $v = $this->encodeProfile($v);
```
Comment (lauripiisang) [inline src/Xhgui/Profiles.php]: why is this recursive call? do we have any other problematic keys than the content of `array_keys($profile['profile'])`?

Comment (lauripiisang) [reply src/Xhgui/Profiles.php]: Are there any nested structures inside a profile that I have ignored?

Comment (lauripiisang) [reply src/Xhgui/Profiles.php]: If this is the case that we do not need recursive checking, we should instead opt for an easier solution.

What do you think about something like 
```php
$replacedKeys = str_replace('.', '．', array_keys($a));
array_combine($replacedKeys, $a);
```
?

Here's a demonstration, if you're intersted

```php
<?php
$a = ['fx' => ['a' => 'xxx', 'b' => 2, 'c' => 3], 'quux' => ['f' => 7, 'cx' => 3]];
print_r(array_combine(str_replace('x', 'y', array_keys($a)), $a));
```

prints

```
Array
(
    [fy] => Array
        (
            [a] => xxx
            [b] => 2
            [c] => 3
        )

    [quuy] => Array
        (
            [f] => 7
            [cx] => 3
        )

)
```

Comment (lauripiisang) [reply src/Xhgui/Profiles.php]: I'd also recommend defining the `.` and `．` as constants (UNICODE_PERIOD and UNICODE_FULLWIDTH_PERIOD) for example?

**Comment by glensc** [comment]
Superseded by https://github.com/perftools/xhgui/pull/352

**Pull request closed without merging**